src/nbl/asset/utils/ISPIRVDebloater.cpp

```diff
@@ -0,0 +1,259 @@
+#include "nbl/asset/utils/ISPIRVDebloater.h"
+#include "nbl/asset/utils/ISPIRVOptimizer.h"
+#include "nbl_spirv_cross/spirv.hpp"
+
+#include "nbl/core/declarations.h"
+#include "nbl/system/ILogger.h"
+#include "spirv-tools/libspirv.hpp"
+
+using namespace nbl::asset;
+
+static constexpr spv_target_env SPIRV_VERSION = spv_target_env::SPV_ENV_UNIVERSAL_1_6;
+
+ISPIRVDebloater::ISPIRVDebloater()
+{
+    constexpr auto optimizationPasses = std::array{
+        ISPIRVOptimizer::EOP_DEAD_BRANCH_ELIM,
+        ISPIRVOptimizer::EOP_ELIM_DEAD_FUNCTIONS,
+        ISPIRVOptimizer::EOP_DEAD_BRANCH_ELIM,
+        ISPIRVOptimizer::EOP_ELIM_DEAD_FUNCTIONS,
```

Comment:
document why you are doing two rounds of same passes https://github.com/Devsh-Graphics-Programming/Nabla/pull/839#discussion_r2046356228